src/ecs/rendering/resources.rs

```diff
@@ -135,3 +137,33 @@ impl AsRef<Rgba> for AmbientColor {
         &self.0
     }
 }
+
+/// This specs resource with id 0 permits sending commands to the
+/// renderer internal window.
+pub struct WindowCommander {
+    // It's unlikely we'll get more than two commands per frame
+    // 2 Boxes also make this the same size as a Vec, so this costs
```

Comment:
How is this true?

Reply:
Every vector is a combination of ptr, length, and capacity.  Each of these is `WORD` size long.  When a SmallVec is inline it does not store a ptr, or capacity, it just stores the given array type.  Box is `WORD` size long so if you make the inline array type (2 * `WORD` size) then even the inline variant doesn't take any more space than it would if it were on the heap.  Structure size isn't bloated, and you still get your smallvec optimization.

Reply:
Oops, right. This comment was stupid.

Reply:
The two-element array somehow made me think it's two times pointer-size.

Reply:
Hmm. So the `SmallVec` copies data from stack to heap if you push more elements than array can hold... Is it a good choice? Why not storing only exceeding amount of elements on heap?
Two boxed traits have size of 4 pointers though.

Reply:
Oh hm I guess that's true.  I'll remove my comment then.  As for storing what you can inline even when it spills, I had the same thought but then you can't operate on a SmallVec as if it were a slice because the memory is not contiguous.